### PR TITLE
Support recursive operator calls for `$sum` aggregation accumulator

### LINKED ISSUE #2694: Support recursive aggregation operators
### What should be done?

After https://github.com/FerretDB/FerretDB/issues/2680

The goal of this task is to support recursive aggregation operators in all supported operators (and accumulators) .

#### Operators support other operators
For example it should be possible to call:
`{$project: {$type: {$type: "foo"}}}`

All of the recursive operators should be tested with compat tests.

It might be necessary to implement better error handling inside `NewOperator` that would return failing operator for the message.

#### Accumulator operator support other operators
- `$group` stage `$sum` accumulator supports operators
example: `totalSaleAmount: { $sum: { $multiply: [ "$price", "$quantity" ] } }` from [manual](https://www.mongodb.com/docs/manual/reference/operator/aggregation/group/#group-by-day-of-the-year)
https://github.com/FerretDB/FerretDB/blob/0992e5f8a410e59041a62a12f666eacf6f3bd27b/internal/handlers/common/aggregations/operators/accumulators/sum.go#L57
Test passes
https://github.com/FerretDB/FerretDB/blob/bddfa9ee15fec28665fdc145bc1d22c7d7f4bf61/integration/aggregate_documents_compat_test.go#L1055-L1064
### Where? 

- `operators` package
- `accumulators` package
See the issue link in the code base.

### Definition of Done

- operators updated;
- integration/compatibility tests added/updated;
- spot refactorings done;
- multiple PRs (separate PRs for different operators).

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #2694.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [ ] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/common/aggregations/operators/accumulators/sum.go [hunk 1]
```diff
@@ -77,7 +95,18 @@ func (s *sum) Accumulate(iter types.DocumentsIterator) (any, error) {
 			return nil, lazyerrors.Error(err)
 		}
 
-		if s.expression != nil {
+		switch {
+		case s.operator != nil:
+			v, err := s.operator.Process(doc)
+			if err != nil {
+				return nil, err
+			}
```
Comment (chilagrow) [inline internal/handlers/common/aggregations/operators/accumulators/sum.go:103]: Let's cover this error case in a test!

**Review by chilagrow — COMMENTED** [review]
Just one tiny comment :hugs:

File: internal/handlers/common/aggregations/operators/accumulators/sum.go (hunk 1 above)
Comment (noisersup) [reply internal/handlers/common/aggregations/operators/accumulators/sum.go:108]: This is unreachable as the Process is called before this line to validate everything. I believe we should refactor that in the next issue and move all of the validations in operators into constructor or separate function though

**Review by chilagrow — APPROVED** [review]
:rocket:

**Review by rumyantseva — APPROVED** [review]
The tests and implementation look good to me!

**Pull request merged**